packages/embed/src/utils/create-iframe/create-iframe.ts

```diff
@@ -14,10 +15,16 @@ export const createIframe = (formId: string, type: EmbedType, options: CreateIfr
   iframe.dataset.testid = 'iframe'
   iframe.addEventListener('load', triggerIframeRedraw, { once: true })
 
-  window.addEventListener('message', getFormReadyHandler(embedId, options))
+  window.addEventListener('message', getFormReadyHandler(embedId, options.onReady))
   window.addEventListener('message', getFormQuestionChangedHandler(embedId, options))
   window.addEventListener('message', getFormSubmitHandler(embedId, options))
 
+  // Get GA instance
+  if (options.shareGoogleAnalyticsInstance) {
+    getFormReadyHandler(embedId, () => {
```

Comment:
I think you need to add listener: `window.addEventListener('message', getFormReadyHandler(...))`

Reply:
Yeah my bad

Reply:
Can we add this to `callbacks.spec.ts` and mock it somehow? My suggestion would be to create a mock GA object in `callbacks.html` and see if correct `postMessage` is sent to iframe (if we can spy on that function, I am not sure).

Reply:
you mean in the functional tests?

Reply:
Yes.

Reply:
I looked into it, I dont think we can listen to messages sent into the iframe. We can proceed without functional tests with just unit tests.

However please test this in production when its merged.

Reply:
Yes I do have one form with the GA id, however the methods are tested with unit test